networking_ccloud/ml2/agent/common/agent.py

```diff
@@ -177,6 +179,21 @@ def apply_config_update(self, context, config):
 
         return result
 
+    @periodic_task.periodic_task(spacing=cfg.CONF.ml2_cc_fabric_agent.persist_config_loop_interval,
+                                 run_immediately=False)
+    def persist_switch_configs(self, context):
+        if not cfg.CONF.ml2_cc_fabric_agent.persist_config_loop_enabled:
+            return
+
+        start_time = time.time()
+        LOG.info("Persisting config on all switches")
+        futures = []
+        for switch in sorted(self._switches, key=attrgetter('name')):
+            futures.append(switch.persist_config())
+        for future in futures:
+            future.result()
```

Comment:
This might raise an exception. Not sure if you don't want to wait for the rest? They will run nonetheless.

Reply:
As we already catch everything in the current implementation of `persist_config()` I think this is fine.